bgpd/bgp_zebra.c

```diff
@@ -1769,6 +1769,89 @@ void bgp_zebra_announce_table_all_subtypes(struct bgp *bgp, afi_t afi,
 						   pi, bgp, afi, safi);
 }
 
+
+/* Announce label change explicit null to implicit null */
+void bgp_zebra_label_set_to_imp_null(struct bgp *bgp, afi_t afi)
+{
+	struct bgp_dest *dest;
+	struct bgp_table *table;
+	struct bgp_path_info *pi;
+
+	/* Don't try to install if we're not connected to Zebra or Zebra doesn't
+	 * know of this instance.
+	 */
+	if (!bgp_install_info_to_zebra(bgp))
+		return;
+
+	table = bgp->rib[afi][SAFI_UNICAST];
+
+	if (!table)
+		return;
+
+	for (dest = bgp_table_top(table); dest; dest = bgp_route_next(dest)) {
```

Comment:
If we have several million routes, shouldn't we be breaking this up into smaller chunks using events?  This will just cpu hog under this condition